clinica/utils/freesurfer.py

```diff
@@ -146,61 +41,182 @@ def generate_regional_measures_alt(
+            write_tsv_file(output_filename, key_list, value_list)
+
+
+def _generate_tsv_for_segmentation(
+    stats_folder: PurePath,
+    output_dir: PurePath,
+    prefix: str,
+) -> None:
+    """Generate TSV files for segmentation."""
+    for filename, suffix in zip(
+        ["aseg.stats", "wmparc.stats"],
+        ["segmentationVolumes.tsv", "parcellation-wm_volume.tsv"],
+    ):
+        stats_filename = stats_folder / filename
+        df = _read_stats_file(stats_filename, "segmentation")
         secondary_stats_dict = get_secondary_stats(stats_filename, "volume")
         key_list = list(df["StructName"]) + list(secondary_stats_dict.keys())
         value_list = list(df["Volume_mm3"]) + list(secondary_stats_dict.values())
-        write_tsv_file(
-            os.path.join(output_dir, prefix + "_parcellation-wm_volume.tsv"),
-            key_list,
-            "volume",
-            value_list,
+        output_filename = output_dir / f"{prefix}_{suffix}"
+        write_tsv_file(output_filename, key_list, value_list)
+
+
+def generate_regional_measures(
+    segmentation_path: str,
+    subject_id: str,
+    atlases: Optional[List[str]] = None,
+    output_dir: Optional[str] = None,
+) -> None:
+    """Read stats files and generate regional measures TSV files.
+
+    The stats files are located in
+    <segmentation_path>/<subject_id>/stats/*.stats
+    and the generated TSV files will be located in the folder
+    <segmentation_path>/regional_measures
+
+    Notes
+    -----
+    The .stats files contain:
+        - a table with statistical information (e.g., structure volume)
+        - secondary statistical information with all lines starting
+          with the sentence '# Measure'.
+
+    The .tsv files return the relevant statistical information from both sources.
+
+    Parameters
+    ----------
+    segmentation_path : str
+        Path to the FreeSurfer segmentation.
+
+    subject_id : str
+        Subject ID in the form `sub-CLNC01_ses-M00`, `sub-CLNC01_long-M00M18` or
+        `sub-CLNC01_ses-M00.long.sub-CLNC01_long-M00M18`.
+
+    atlases : List[str], optional
+        The atlas names to be used.
+        By default, the following atlases will be used:
+            - "desikan"
+            - "destrieux"
+            - "ba"
+
+    output_dir : str, optional
+        Path to the folder where the .tsv stats files will be stored.
+        If no output folder is provided, files will be written to
+        [path_segmentation]/regional_measures.
+    """
+    import errno
+    import os
+    from pathlib import Path
+
+    atlases = atlases or ["desikan", "destrieux", "ba"]
+    prefix = _get_prefix(subject_id)
+    seg_path = Path(os.path.expanduser(segmentation_path))
+    stats_folder = seg_path / subject_id / "stats"
+
+    if not stats_folder.is_dir():
+        raise IOError(
+            "Image %s does not contain FreeSurfer segmentation"
+            % prefix.replace("_", " | ")
```

Comment:
I think [FileNotFoundError](https://docs.python.org/3/library/exceptions.html#FileNotFoundError) would be more appropriate.

```suggestion
        raise FileNotFoundError(
            "Image %s does not contain FreeSurfer segmentation"
            % prefix.replace("_", " | ")
```